citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -494,4 +498,26 @@ public CitationPage getPrevAndNextCitation(IUser user, String groupId, String co
     public void deleteLocalGroupCitations(String groupId) {
         citationStore.deleteCitationByGroupId(groupId);
     }
+
+	@Override
+	public ICitation updateCitationReference(ICitation citation, String reference) {
+		Set<IReference> references = citation.getReferences();
+        Boolean referenceExists = false;
+        if (references == null) {
+            references = new HashSet<>();
+        } else {
+            for (IReference refer : references) {
+                if (refer.getReferenceString().equals(reference)) {
```

Comment:
this should not be checked. the only thing that could be checked would be the citation id, to not have duplicate links to the same citation.